include/nlohmann/detail/output/serializer.hpp

```diff
@@ -664,6 +664,19 @@ class serializer
         }
     }
 
+    // templates to avoid warnings about useless casts
+    template <typename NumberType, enable_if_t<std::is_signed<NumberType>::value, int> = 0>
+    bool is_negative_number(NumberType x)
```

Comment:
```
template <typename NumberType>
enable_if_t<std::is_signed<NumberType>::value, bool> is_negative_number(NumberType x)
```

Reply:
?

Reply:
Try this. You use legal but verbose and very indirect way for `enable_if_t` utilize.

Reply:
That is a normal way to do it for functions that are already templates.   https://en.cppreference.com/w/cpp/types/enable_if

Reply:
"legal but verbose", Ok
And how about this?
https://github.com/nlohmann/json/blob/700b95f4473cac1d900b807541daad0ca3d98f52/include/nlohmann/detail/conversions/from_json.hpp#L421-L423

Reply:
I'm not sure why @theodelrieu used the longer form here compared to the rest of the file, perhaps there was a technical reason.  If not, it would make sense to use the value form as in the rest of the file. https://github.com/nlohmann/json/commit/c5e63fd68430469f3734b0468b1db2e9d4fc5674